src/Builder/Accumulator/AccumulatorAccumulator.php

```diff
@@ -70,20 +70,36 @@ public function __construct(
         Optional|PackedArray|ResolvesToArray|BSONArray|array $initArgs = Optional::Undefined,
         Optional|Javascript|string $finalize = Optional::Undefined,
     ) {
+        if (is_string($init)) {
+            $init = new Javascript($init);
+        }
+
         $this->init = $init;
+        if (is_string($accumulate)) {
+            $accumulate = new Javascript($accumulate);
```

Comment:
The [`MongoDB\BSON\Javascript`](https://www.php.net/manual/en/class.mongodb-bson-javascript.php) constructor accepts a 2nd parameter `$scope`. I don't think we need to set anything.

Reply:
The scope has been deprecated since 4.4 ([SERVER-45460](https://jira.mongodb.org/browse/SERVER-45460)), so no need to support it.

Reply:
Thanks, that's what I just concluded by reading the [list of types](https://www.mongodb.com/docs/manual/reference/operator/query/type/#available-types).